README.md

```diff
@@ -6,21 +6,22 @@ Oval is a data validation tool for S3-compatible object storages.
 
 ## Motivation
 
-Storage systems can hardly achieve 100% of data integrity. 
-Though there is a lot of possible cause of data corruption, the data corruption due to software bugs is catastrophic. So continuous efforts to reduce the possibility of data corruption are required.   
+Storage systems can hardly achieve 100% of data integrity. Data is usually damaged by two factors; failures and software bugs.
+For failures, there are many techniques to reduce the impact (RAID, erasure coding, backup etc.). Though it is impossible to protect storage systems perfectly from data loss due to failures, the impact of failures can be controlled by paying the cost for data redundancy.
 
-Some storage systems have an internal defense mechanism against data corruption. However, it is sometimes useless when the data get corrupted in undetectable ways, or if there are bugs in the defense mechanism itself. In that case, data will get corrupted silently.
+On the other hand, the data corruption due to software bugs is unacceptable. It is hard to estimate the damage of data loss due to bugs because bugs can cause any bad things. For example, some bugs may destroy all redundant data copies. Once you start to worry about bugs, you cannot stop it. Thus, continuous efforts to reduce data corruption bugs are required.
 
-So, it is important for storage system's developers and users
-to check if their storage systems have enough data integrity from the storage user's point of view.
-People sometimes check data integrity by the following steps.
+Some storage systems have internal defense mechanisms against data corruption (periodic CRC check, online T10 DIF check etc.). However, these mechanisms are never perfect for two reasons. First, they have theoretical limit to detect data corruption. Secondly, these mechanisms may also have bugs since they are also implemented by humans.
```

Comment:
```suggestion
Some storage systems have internal defense mechanisms against data corruption (periodic CRC check, synchronus T10 DIF check etc.). However, these mechanisms are never perfect for two reasons. First, they have theoretical limit to detect data corruption. Secondly, these mechanisms may also have bugs since they are also implemented by humans.
```